File: javascript/resistor-color-duo/resistor-color-duo.js
const COLORS = [
  "black",
  "brown",
  "red",
  "orange",
  "yellow",
  "green",
  "blue",
  "violet",
  "grey",
  "white"
]

export const value = (arr) => {
  const total = arr.slice(0,2).map(color => COLORS.findIndex(item => item === color)).join("")
  return Number(total);
};

File: ruby/resistor-color-duo/resistor_color_duo.rb
class ResistorColorDuo
  COLORS = %w[black brown red orange yellow green blue violet grey white].freeze

  class << self
    def value(args)
      args[0..1].map do |a|
        COLORS.index(a)
      end.join().to_i
    end
  end
end
